src/bsp/platform/stm32fx/xi_bsp_time_stm32fx.c

```diff
@@ -13,17 +13,23 @@
 #endif
 #include <xi_bsp_time_stm32fx_sntp.h>
 
+static uint32_t ticks_at_start = 0;
+
 void xi_bsp_time_init()
 {
     xi_bsp_time_sntp_init( NULL );
+    if (xi_bsp_time_sntp_getseconds_posix() != 0)
+    {
```

Comment:
Why is this condition necessary?
If the `getseconds` equals zero, then isn't it a failure? Failure during getting the time from SNTP server? Also means the `HAL_GetTick()` returns zero, right?

If time isn't received from SNTP service the connection to broker should fail anyway. So, I would stick to a simple:

`ticks_at_start = HAL_GetTick();` without a condition.